prometheus/src/cluster_metrics.rs

```diff
@@ -1,12 +1,84 @@
 use solana_gossip::cluster_info::ClusterInfo;
 use solana_runtime::bank::Bank;
+use solana_sdk::{clock::Slot, pubkey::Pubkey};
+use solana_vote_program::vote_state::VoteState;
 
 use crate::{
     token::Lamports,
     utils::{write_metric, Metric, MetricFamily},
 };
 use std::{io, sync::Arc, time::SystemTime};
 
+struct ValidatorVoteInfo {
+    vote_address: Pubkey,
+    balance: Lamports,
+    last_vote: Slot,
+}
+
+impl ValidatorVoteInfo {
+    fn new_from_bank(bank: &Arc<Bank>, identity_pubkey: &Pubkey) -> Option<Self> {
+        let vote_accounts = bank.vote_accounts();
+        let vote_state_default = VoteState::default();
+        vote_accounts
+            .iter()
+            .filter_map(|(&vote_pubkey, (_activated_stake, account))| {
+                let vote_state = account.vote_state();
+                let vote_state = vote_state.as_ref().unwrap_or(&vote_state_default);
+                if identity_pubkey != &vote_state.node_pubkey {
+                    return None;
+                }
+                let last_vote = if let Some(vote) = vote_state.votes.iter().last() {
+                    vote.slot
+                } else {
+                    0
+                };
+                let vote_balance = Lamports(bank.get_balance(&vote_pubkey));
+                Some(ValidatorVoteInfo {
+                    vote_address: vote_pubkey,
+                    balance: vote_balance,
+                    last_vote,
+                })
+            })
+            .next()
+    }
+
+    fn write_prometheus<W: io::Write>(&self, out: &mut W, at: SystemTime) -> io::Result<()> {
+        write_metric(
+            out,
+            &MetricFamily {
+                name: "solana_cluster_vote_public_key_info",
+                help: "The current Solana node's vote public key",
+                type_: "count",
+                metrics: vec![Metric::new(1)
+                    .with_label("vote", self.vote_address.to_string())
+                    .at(at)],
+            },
+        )?;
+        // We can use this metric to track if the validator is making progress
+        // by voting on the last slots.
+        write_metric(
+            out,
+            &MetricFamily {
+                name: "solana_cluster_last_vote_slot_count",
+                help: "The last slot that the validator voted on",
```

Comment:
```suggestion
                help: "The voted-on slot of the validator's last vote that got included in the chain",
```

Also, if I understand correctly, this is the last voted on slot, *from the point of view of the chain* (because we access it through the vote account from the bank), *as seen from the validator* (because inside the validator, we only know our own view of the chain). So the sequence is:

1. We vote on slot _x_.
2. The leader includes that vote in block _y_.
3. We receive block _y_.

Only at step 3 would we observe that we voted, and only if we are synced to the chain and using the right fork. This is an important metric to have, but it is slightly different from the the last vote we _sent_ (which probably we cannot get from just the bank).

Because it is based on the bank, I think we should split this metric into the different commitment levels as well, like the other bank-based metrics.